home/utils.py

```diff
@@ -217,86 +248,338 @@ def add_viber(row, page=None):
+        page.subtitle,
+        str(page.body),
+        page.whatsapp_title,
+        messenging_platform_messages["whatsapp"][0]["msg"]
+        if len(messenging_platform_messages["whatsapp"]) == 1
+        else "",
+        page.messenger_title,
+        messenging_platform_messages["messenger"][0]["msg"]
+        if len(messenging_platform_messages["messenger"]) == 1
+        else "",
+        page.viber_title,
+        messenging_platform_messages["viber"][0]["msg"]
+        if len(messenging_platform_messages["viber"]) == 1
+        else "",
+        str(page.translation_key),
+        format_list_from_query_set(page.tags.all()),
+        format_list_from_query_set(page.quick_replies.all()),
+        format_list_from_query_set(page.triggers.all()),
+        str(page.locale),
+        get_next_prompts(messenging_platform_messages, 0),
+        get_image_link(messenging_platform_messages, 0),
+        get_doc_link(messenging_platform_messages, 0),
+        get_media_link(messenging_platform_messages, 0),
+        format_list_from_query_set(get_related_pages(page)),
+    ]
+    if most_messages == 0:
+        # if a page has no content
+        basic_row[5] = 0
+        return [basic_row]
+
+    for message_index in range(most_messages):
+        if message_index == 0:
+            row = basic_row
+        else:
+            row = padding + [""] * len(fieldnames)
+            row[21] = get_next_prompts(messenging_platform_messages, message_index)
+            row[22] = get_image_link(messenging_platform_messages, message_index)
+            row[23] = get_doc_link(messenging_platform_messages, message_index)
+            row[24] = get_media_link(messenging_platform_messages, message_index)
+        row[5] = message_index + 1
+
+        if message_index < len(messenging_platform_messages["whatsapp"]):
+            row[11] = messenging_platform_messages["whatsapp"][message_index]["msg"]
+
+        if message_index < len(messenging_platform_messages["messenger"]):
+            row[13] = messenging_platform_messages["messenger"][message_index]["msg"]
+
+        if message_index < len(messenging_platform_messages["viber"]):
+            row[15] = messenging_platform_messages["viber"][message_index]["msg"]
+        rows.append(row)
+    return rows
+
+
+def add_children(temp_sheet: List[list], children: PageQuerySet) -> List[list]:
+    """Recursive function that traverses the children of a page with a depth first search algorithm"""
+    for child in children:
+        content_page = ContentPage.objects.filter(id=child.id).first()
+        row = get_rows(content_page)
+        temp_sheet.extend(row)
+        if content_page.has_children:
+            add_children(temp_sheet, content_page.get_children())
+    return temp_sheet
+
+
+def get_content_sheet() -> List[list]:
+    content_sheet = []
+    headings = [1, 2, 3, 4, 5, "Message"] + fieldnames
+    content_sheet.append(headings)
+    home = HomePage.objects.filter(locale_id=1).first()
```

Comment:
I don't know wagtail well enough for this, but is there not a kind of home local, or default or root locale, that we can rely on instead of hardcoding the primary key here? The issue with something like this is if you add a row, then delete it, then add a row again, then that key doesn't get recycled, so you won't ever get a row with the ID of 1 again.

Reply:
Oh yes, so I actually did not mean to hardcode this but I also don't know enough about wagtail to figure it out so I meant to ask Codie but it slipped my mind :) thank you! @codiebeulaine could you advise?